### ISSUE TITLE
Feature request: Connect Docker instance to multiple networks

### ISSUE THREAD

**Issue by marcoreni** [aws/aws-sam-cli#7559]
### Describe your idea/feature/enhancement

In some complex scenarios (eg. using localstack with some other dependency) it may be useful to connect the Lambda with multiple Docker Networks.

Docker already supports the functionality of connecting one container to multiple networks, so it would just be a matter of exposing a new option so that a list of networks can be provided.

### Proposal

Expose `--docker-networks` (or accept multiple `--docker-network` values).

**Comment by mndeveci** [comment]
Thanks for the recommendation @marcoreni,

`sam local` commands already exposes `--docker-network` parameter (see one example [here](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-cli-command-reference-sam-local-invoke.html)), I assume your intention to define it multiple times so that it can be attached to multiple docker networks?

**Comment by marcoreni** [comment]
That's correct! We're trying to use localstack but at the same time we have some external services that we'd like to connect to our lambdas. Right now we need to pass through host networking, but having multiple networks connected would be ideal.
